Explorer/Assets/DCL/InWorldCamera/CameraReelGallery/Components/ReelThumbnailController.cs

```diff
@@ -0,0 +1,121 @@
+using Cysharp.Threading.Tasks;
+using DCL.InWorldCamera.CameraReelStorageService;
+using DCL.InWorldCamera.CameraReelStorageService.Schemas;
+using DG.Tweening;
+using System;
+using System.Threading;
+using UnityEngine;
+using Utility;
+
+namespace DCL.InWorldCamera.CameraReelGallery.Components
+{
+    public class ReelThumbnailController : IDisposable
+    {
+        internal readonly ReelThumbnailView view;
+        private readonly ICameraReelScreenshotsStorage cameraReelScreenshotsStorage;
+
+        private OptionButtonController optionButton;
+        private CancellationTokenSource loadImageCts;
+
+        public event Action<CameraReelResponseCompact, Sprite>? ThumbnailLoaded;
+        public event Action<CameraReelResponseCompact>? ThumbnailClicked;
+
+        public CameraReelResponseCompact CameraReelResponse { get; private set; }
+
+        public ReelThumbnailController(ReelThumbnailView view,
+            ICameraReelScreenshotsStorage cameraReelScreenshotsStorageService)
+        {
+            this.view = view;
+            this.cameraReelScreenshotsStorage = cameraReelScreenshotsStorageService;
+            this.view.PointerEnter += OnPointerEnter;
+            this.view.PointerExit += OnPointerExit;
+        }
+
+        public void Setup(CameraReelResponseCompact cameraReelData, OptionButtonController optionsButton)
+        {
+            this.CameraReelResponse = cameraReelData;
+            this.optionButton = optionsButton;
+
+            if (this.optionButton is not null)
```

Comment:
Can it be null?

Reply:
same reason as for the previous conversation